packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -282,7 +282,14 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
         event is PointerAddedEvent ||
         event is PointerRemovedEvent) {
       assert(event.position != null);
-      _mouseTracker!.updateWithEvent(event, () => hitTestResult ?? renderView.hitTestMouseTrackers(event.position));
+      _mouseTracker!.updateWithEvent(
```

Comment:
Could we only call `_mouseTracker!.updateWithEven` when it is a mouse event?

Reply:
+1

Reply:
The idea is that `RendererBinding` provides all events and a way to compute hit test result for `MouseTracker` to retrieve when needed. In this way most logic is hidden in `MouseTracker`. Nevertheless I agree that the hit testing function is a little cumbersome. I'll take a moment to think if there's a better way but I'm probably sticking to it in the end.

Reply:
It turns out that the `if` clause is unnecessary at all (according to the documentation, which I trust). Several assertions can also be removed since the NNBD. This should feel much cleaner.